src/todo_list/core.clj

```diff
@@ -135,34 +135,53 @@
                             [:td kind]
                             [:td estimate]]))))]))))
 
+(defn not-found
+  [_req]
+  (-> (resp/response
+        (html5 {}
+          [:h1 "This is not the page you are looking for"]
+          [:p "Sorry, the page you requested was not found!"]))
+      (resp/status 404)))
+
 ;---------------------------------------------------------------------------
 
-(defroutes app
+(def ROUTES
+  ["/" [["" {:get ::home}]
+        ["req-info" {:get ::req-info}]
+        [["calculator/" [#"[\+\-\:\*]" :op] "/" [#"\d+" :a] "/" [#"\d+" :b]] {:get ::calculator}]
+        ["mytasks" {:get ::tasks}]
+        [["diff/" [#".+" :poly]] {:get ::diff}]
+        [true ::not-found]]])
+
+(comment
+  (bidi/match-route ROUTES "/mytasks")
+  (bidi/path-for ROUTES ::calculator :op "+" :a "1" :b "2")
+  (bidi/path-for ROUTES ::diff :args "1/2/")
+  )
+
+(def HANDLERS
+  {::home       the-handler
+   ::req-info   req-info
+   ::calculator calculator
+   ::diff       thediff
+   ::tasks      tasks
+   ::not-found  not-found})
```

Comment:
Compojure is quite a dirty library (plus I don't know it well).  Switched you over to bidi, got everything hooked up and routing properly.